optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py

```diff
@@ -366,7 +366,7 @@ def __call__(
                 "Passing `callback_steps` as an input argument to `__call__` is deprecated, consider use `callback_on_step_end`",
             )
 
-        with torch.autocast(device_type="hpu", dtype=torch.bfloat16, enabled=self.gaudi_config.use_torch_autocast):
+        with torch.autocast(device_type="hpu", dtype=torch.bfloat16, enabled=(self.dtype != torch.float)):
```

Comment:
We cannot do that. Not everyone wants to run autocast. For example, if we use the `bf16_full_eval` arg, the whole model is in bf16 and in that case we don't want to use autocast. The user should have the choice to use it or not.

Reply:
answer as below